language/tools/disassembler/src/main.rs

```diff
@@ -88,18 +91,23 @@ fn main() {
 
     // TODO: make source mapping work with the Move source language
     let no_loc = Spanned::unsafe_no_loc(()).loc;
+    let module: CompiledModule;
+    let script: CompiledScript;
```

Comment:
Out of curiosity, I'm guessing these variables need to be declared here to make the lifetimes work out?

Reply:
Exactly right. I don't know if there's a better way to write this in Rust, but declaring them within the `if` branches would cause them to go out of scope and be freed. The `BinaryIndexedView` only holds a reference to them and would outlive them if they were freed within those branches.